python/Ganga/Utility/logging/__init__.py

```diff
@@ -137,36 +176,67 @@ def _make_file_handler(logfile, logfile_size):
             try:
                 # WARNING: this relies on the implementation details of the
                 # logging module
-                del logging._handlers[file_handler]
+                del _get_logging()._handlers[file_handler]
             except KeyError:
                 # don't complain if the handler was correctly unregistered by
                 # the logging system
                 pass
 
-        new_file_handler.setFormatter(logging.Formatter(_formats['VERBOSE']))
+        new_file_handler.setFormatter(_get_logging().Formatter(_formats['VERBOSE']))
         main_logger.addHandler(new_file_handler)
         file_handler = new_file_handler
 
+# set the loglevel for a logger to a given string value (example: "DEBUG")
+def _set_log_level(logger, value):
+
+    global _global_level
+
+    if _global_level is not None:
+        value = _global_level
+
+    # convert a string "DEBUG" into enum object logging.DEBUG
+    def _string2level(name):
+        if hasattr(logging, name):
+            return getattr(logging, name)
+
+    try:
+        logger.setLevel(_string2level(value))
+        #if value != 'DEBUG':
+        #    return value
+        #    #logging._srcfile = None
+        #    #logging.logThreads = 0
+        #    #logging.logThreads = 0
+        return value
+    except AttributeError as err:
+        logger.error('Attribute Error: %s', str(err))
+        logger.warning('possible configuration error: invalid level value (%s), using default level', value)
+        return None
+
 
 # reflect all user changes immediately
 def post_config_handler(opt, value):
-
-    if '_customFormat' in config and config['_customFormat'] != "":
+    global config, _formats
```

Comment:
Do these need to `global`. If we're not rebinding them then the name lookup should find the correct objects anyway?